File: rudolph/src/main/java/cn/wzbos/android/rudolph/Interceptor.java
package cn.wzbos.android.rudolph;


import android.content.Context;

import cn.wzbos.android.rudolph.router.Router;


/**
 * Created by wuzongbo on 2017/4/11.
 */
public interface Interceptor {
    boolean intercept(Context context, Router<?> router);
}

File: rudolph/src/main/java/cn/wzbos/android/rudolph/Interceptor.kt
package cn.wzbos.android.rudolph

import android.content.Context
import cn.wzbos.android.rudolph.router.Router

/**
 * Created by wuzongbo on 2017/4/11.
 */
interface Interceptor {
    fun intercept(context: Context?, router: Router<*>): Boolean
}